### DISCUSSION TITLE
how to use in nodeJS

### DISCUSSION THREAD

**Discussion by dajun-TT** [CoderLine/alphaTab#1300] [General]
Hi! I tried using AlphaTab in Node JS. I was able to load the scores, displaying score title, artist, tracks...but when using renderer functions like (partialLayoutFinished, partialRenderFinished...) copied from docs, renderResult was always null. I don't know if it's a bug or am I doing something completely wrong.
I followed the example given by official website， it  could normally run, but I cannot get the SVG String when I use those functions like partialLayoutFinished...

**Comment by Danielku15** [comment]
Hi. 
Can you share the code you are using currently?  One special thing which is currently incorrect in the docs is that you cannot call `renderResult` directly in the `partialLayoutFinished` but need to delay the execution:

```js
renderer.partialLayoutFinished.on(r => {
    setImmediate(()=> {
        renderer.renderResult(r.id);
    })
});
```

or as alternative disable the lazy loading via `settings.core.enableLazyLoading = false;`
